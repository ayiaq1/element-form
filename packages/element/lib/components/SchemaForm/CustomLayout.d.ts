import React from 'react';
import type { FormProps } from 'antd/lib/form';
import type { ISchemaProps } from './typings';
declare const _default: React.MemoExoticComponent<
  ({
    type,
    options,
    disabled: formDisabled,
    form,
    ...formReset
  }: FormProps<any> & ISchemaProps) => JSX.Element
>;
export default _default;
