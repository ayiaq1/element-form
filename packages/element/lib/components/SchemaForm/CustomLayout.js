var __assign =
  (this && this.__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];
          for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
      };
    return __assign.apply(this, arguments);
  };
var __rest =
  (this && this.__rest) ||
  function (s, e) {
    var t = {};
    for (var p in s)
      if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === 'function')
      for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
          t[p[i]] = s[p[i]];
      }
    return t;
  };
define(['require', 'exports', 'react', './ItemWrap', './ItemGroupWrap'], function (
  require,
  exports,
  react_1,
  ItemWrap_1,
  ItemGroupWrap_1,
) {
  'use strict';
  Object.defineProperty(exports, '__esModule', { value: true });
  var CustomLayout = function (_a) {
    var _b = _a.type,
      type = _b === void 0 ? 'custom' : _b,
      _c = _a.options,
      options = _c === void 0 ? [] : _c,
      formDisabled = _a.disabled,
      form = _a.form,
      formReset = __rest(_a, ['type', 'options', 'disabled', 'form']);
    return react_1.default.createElement(
      react_1.default.Fragment,
      null,
      options.map(function (item, index) {
        var _a, _b, _c;
        return react_1.default.createElement(
          react_1.Fragment,
          {
            key:
              (_b =
                (_a = item === null || item === void 0 ? void 0 : item.name) === null ||
                _a === void 0
                  ? void 0
                  : _a.toString()) !== null && _b !== void 0
                ? _b
                : index.toString(),
          },
          item.type === 'group'
            ? react_1.default.createElement(
                ItemGroupWrap_1.default,
                __assign({}, item, {
                  formDisabled: formDisabled,
                  initialValues: formReset.initialValues,
                }),
              )
            : react_1.default.createElement(
                ItemWrap_1.default,
                __assign({}, item, {
                  values:
                    (item === null || item === void 0 ? void 0 : item.name) &&
                    ((_c =
                      formReset === null || formReset === void 0
                        ? void 0
                        : formReset.initialValues) === null || _c === void 0
                      ? void 0
                      : _c[item === null || item === void 0 ? void 0 : item.name]),
                  formDisabled: formDisabled,
                  initialValues: formReset.initialValues,
                }),
              ),
        );
      }),
    );
  };
  exports.default = (0, react_1.memo)(CustomLayout);
});
//# sourceMappingURL=CustomLayout.js.map
