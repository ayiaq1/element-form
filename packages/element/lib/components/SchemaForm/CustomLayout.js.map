{"version":3,"file":"CustomLayout.js","sourceRoot":"","sources":["../../../src/components/SchemaForm/CustomLayout.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;IAYA,IAAM,YAAY,GAAG,UAAC,EAMK;QALzB,IAAA,YAAe,EAAf,IAAI,mBAAG,QAAQ,KAAA,EACf,eAAY,EAAZ,OAAO,mBAAG,EAAE,KAAA,EACF,YAAY,cAAA,EACtB,IAAI,UAAA,EACD,SAAS,cALQ,uCAMrB,CADa;QAEZ,OAAO,CACL,8DACG,OAAO,CAAC,GAAG,CAAC,UAAC,IAAe,EAAE,KAAK;;YAClC,OAAO,CACL,8BAAC,gBAAQ,IAAC,GAAG,EAAE,MAAA,MAAA,IAAI,aAAJ,IAAI,uBAAJ,IAAI,CAAE,IAAI,0CAAE,QAAQ,EAAE,mCAAI,KAAK,CAAC,QAAQ,EAAE,IACtD,IAAI,CAAC,IAAI,KAAK,OAAO,CAAC,CAAC,CAAC,CACvB,8BAAC,uBAAa,eAAK,IAAI,IAAE,YAAY,EAAE,YAAY,EAAE,aAAa,EAAE,SAAS,CAAC,aAAa,IAAI,CAChG,CAAC,CAAC,CAAC,CACF,8BAAC,kBAAQ,eACH,IAAI,IACR,MAAM,EAAE,CAAA,IAAI,aAAJ,IAAI,uBAAJ,IAAI,CAAE,IAAI,MAAI,MAAA,SAAS,aAAT,SAAS,uBAAT,SAAS,CAAE,aAAa,0CAAG,IAAI,aAAJ,IAAI,uBAAJ,IAAI,CAAE,IAAW,CAAC,CAAA,EACnE,YAAY,EAAE,YAAY,EAC1B,aAAa,EAAE,SAAS,CAAC,aAAa,IACtC,CACH,CACQ,CACZ,CAAC;QACJ,CAAC,CAAC,CACD,CACJ,CAAC;IACJ,CAAC,CAAC;IACF,kBAAe,IAAA,YAAI,EAAC,YAAY,CAAC,CAAC"}