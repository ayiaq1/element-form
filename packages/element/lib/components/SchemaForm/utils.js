/*
 * @Author: yihuang
 * @Date: 2022-03-01 11:47:38
 * @Description: 转换工具
 * @LastModifiedBy: yihuang
 */
/** 转换所有UI组件的onChange数据 */
export var converChangeEvent = function (params) {
  var str;
  if (params === null || params === void 0 ? void 0 : params.target) {
    str = params.target.value;
  } else {
    str = params;
  }
  return str;
};
