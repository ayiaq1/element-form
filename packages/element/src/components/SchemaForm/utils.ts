/*
 * @Author: yihuang
 * @Date: 2022-03-01 11:47:38
 * @Description: 转换工具
 * @LastModifiedBy: yihuang
 */

/** 转换所有UI组件的onChange数据 */
export const converChangeEvent = (params: any): any => {
  let str;
  if (params?.target) {
    str = params.target.value;
  } else {
    str = params;
  }
  return str;
};
