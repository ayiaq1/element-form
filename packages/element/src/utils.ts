import {
  FormatTimeType,
  Unix,
  UnixValueOf,
  MOMENT,
  HMS,
  DataType,
  IConverTimeType,
} from './typings';
import moment, { isMoment } from 'moment';
// 转换为 日期的格式化值
const formatMomentFunc = (item: any, format: FormatTimeType, convertResult?: IConverTimeType) => {
  switch (format) {
    case Unix:
      // 时间戳 秒
      return isMoment(item) ? item.unix() : moment(item).unix();
    case UnixValueOf:
      // 时间戳 毫秒
      return isMoment(item) ? item.valueOf() : moment(item).valueOf();
    case MOMENT:
      // moment
      return isMoment(item) ? item : moment(item);
    case HMS:
      return isMoment(item) ? item.format(HMS) : moment(item, HMS).format(HMS);
    default:
      if (typeof item === 'string') {
        return item;
      } else if (isMoment(item)) {
        return item.format((convertResult as DataType) ?? format);
      } else if (Number.isInteger(item)) {
        return moment(item).unix();
      } else {
        return moment(item).format(format);
      }
    // return typeof item === 'string'
    //   ? item
    //   : isMoment(item)
    //   ? item.format((convertResult as DataType) ?? format)
    //   : Number.isInteger(item)
    //   ? moment.unix(item)
    //   : moment(item).format(format);
  }
};
export { formatMomentFunc };
