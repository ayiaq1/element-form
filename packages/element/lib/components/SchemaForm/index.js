var __assign =
  (this && this.__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];
          for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
      };
    return __assign.apply(this, arguments);
  };
var __rest =
  (this && this.__rest) ||
  function (s, e) {
    var t = {};
    for (var p in s)
      if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === 'function')
      for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
          t[p[i]] = s[p[i]];
      }
    return t;
  };
import React, { memo } from 'react';
import { Form } from 'antd';
import RowLayout from './RowLayout';
import CustomLayout from './CustomLayout';
var SchemaForm = function (_a) {
  var _b = _a.type,
    type = _b === void 0 ? 'row' : _b,
    resetProps = __rest(_a, ['type']);
  var _c = resetProps.options,
    options = _c === void 0 ? [] : _c,
    form = resetProps.form,
    formReset = __rest(resetProps, ['options', 'form']);
  return React.createElement(
    React.Fragment,
    null,
    (options === null || options === void 0 ? void 0 : options.length)
      ? React.createElement(
          Form,
          __assign({}, formReset, { form: form }),
          type === 'row'
            ? React.createElement(RowLayout, __assign({ type: type }, resetProps))
            : React.createElement(CustomLayout, __assign({ type: type }, resetProps)),
        )
      : null,
  );
};
SchemaForm.displayName = 'SchemaForm';
export default memo(SchemaForm);
