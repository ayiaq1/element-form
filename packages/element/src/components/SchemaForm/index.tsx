import React, { memo } from 'react';
import { Form } from 'antd';
import { FormProps } from 'antd/lib/form';
import { ISchemaProps } from './typings';
import RowLayout from './RowLayout';
import CustomLayout from './CustomLayout';

const SchemaForm = ({ type = 'row', ...resetProps }: FormProps & ISchemaProps) => {
  const { options = [], form, ...formReset } = resetProps;
  return (
    <>
      {options?.length ? (
        <Form {...formReset} form={form}>
          {type === 'row' ? (
            <RowLayout type={type} {...resetProps} />
          ) : (
            <CustomLayout type={type} {...resetProps} />
          )}
        </Form>
      ) : null}
    </>
  );
};
SchemaForm.displayName = 'SchemaForm';
export default memo(SchemaForm);
