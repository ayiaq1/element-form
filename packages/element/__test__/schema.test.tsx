/*
 * @Author: yihuang
 * @Date: 2022-03-23 16:29:31
 * @Description: schema 组件
 * @LastModifiedBy: yihuang
 */
import React, { useState } from 'react';
import { mount } from 'enzyme';
import type { IFormItem } from '../src/components/SchemaForm/typings';
import renderer from 'react-test-renderer';
import SchemaForm from '../src/components/SchemaForm';
import mountTest from '../tests/mountTest';
import { Form, Button, Space } from 'antd';
import { subtractRangeDays, waitForComponentToPaint } from '../tests/utils';

const TextItem: IFormItem = {
  label: 'text',
  name: 'text',
  type: 'text',
};

const InputItem: IFormItem = {
  label: 'input',
  name: 'input',
  type: 'input',
};
const IDateItem: IFormItem = {
  label: 'rangepicker',
  name: 'rangepicker',
  type: 'rangepicker',
};
// Element 组件
describe('SchemaForm Element', () => {
  mountTest(SchemaForm);

  it('Save type is row snapshot', async () => {
    const wrapper = renderer.create(<SchemaForm type="row" options={[TextItem, InputItem]} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('Save type is custom snapshot', async () => {
    const wrapper = renderer.create(<SchemaForm type="custom" options={[TextItem, InputItem]} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('initialValues set props', async () => {
    const initialValues = {
      text: 'text',
      input: 'input',
    };
    const Demo = () => {
      const [form] = Form.useForm();
      return (
        <SchemaForm
          disabled
          form={form}
          type="custom"
          initialValues={initialValues}
          options={[TextItem, InputItem]}
        />
      );
    };
    const wrapper = mount(<Demo />);
    const formItems = wrapper.find('.ant-row.ant-form-item');
    expect(formItems.length).toBe(2);
    expect(formItems.at(0).find('label').text()).toBe(initialValues.text);
    expect(formItems.at(1).find('input').prop('value')).toBe(initialValues.input);
    expect(formItems.at(1).find('input').first().prop('value')).toBeTruthy();
  });

  it('initialValues reset', async () => {
    const onValuesChange = jest.fn();
    const onResetFunc = jest.fn();
    const onFinishFunc = jest.fn();
    const initialValues = {
      text: 'text',
      input: 'input',
      rangepicker: subtractRangeDays(7),
    };
    const ChangeValues = {
      text: 'text1',
      input: 'input2',
      rangepicker: subtractRangeDays(1),
    };
    const Demo = () => {
      const [form] = Form.useForm();
      const onResetClick = () => {
        form.resetFields();
        onResetFunc.mockReturnValue(form.getFieldsValue());
      };
      const onSubmitClick = () => {
        onFinishFunc.mockReturnValue(form.getFieldsValue());
      };
      const onChangeClick = () => {
        form.setFieldsValue({ ...ChangeValues });
      };
      const ResetItem: IFormItem = {
        type: 'custom',
        render: () => (
          <Space>
            <Button className="reset-btn" onClick={onResetClick}>
              重置
            </Button>
            <Button className="change-btn" onClick={onChangeClick}>
              修改数据
            </Button>
            <Button className="submit-btn" onClick={onSubmitClick}>
              提交查看数据
            </Button>
          </Space>
        ),
      };
      return (
        <SchemaForm
          form={form}
          type="custom"
          initialValues={initialValues}
          options={[TextItem, InputItem, IDateItem, ResetItem]}
          onValuesChange={onValuesChange}
        />
      );
    };
    const wrapper = mount(<Demo />);
    // 修改内容
    wrapper.find('.change-btn').first().simulate('click');
    await waitForComponentToPaint(wrapper);
    wrapper.find('.submit-btn').first().simulate('click');
    await waitForComponentToPaint(wrapper);
    expect(onFinishFunc()).toEqual(ChangeValues);
    // 重置数据
    wrapper.find('.reset-btn').first().simulate('click');
    await waitForComponentToPaint(wrapper);
    // 重置之后，数据应该为初始值
    expect(onResetFunc()).toEqual(initialValues);
  });
});
// 自定义节点
describe('SchemaForm Custom', () => {
  const onValuesChange = jest.fn();
  const Demo = () => {
    const [form] = Form.useForm();
    const [formDisabled, setFormDisabled] = useState(true);
    return (
      <SchemaForm
        disabled={formDisabled}
        form={form}
        type="custom"
        initialValues={{
          diy: 'diy-value',
        }}
        onValuesChange={onValuesChange}
        options={[
          {
            label: '自定义组件',
            name: 'diy',
            type: 'custom',
            render: ({ disabled, value, onChange }) => {
              return (
                <Space>
                  <Button className="change-btn" disabled={disabled} onClick={onChange}>
                    {value}
                  </Button>
                  <Button className="cancel-btn" onClick={() => setFormDisabled(false)}>
                    取消禁用
                  </Button>
                </Space>
              );
            },
          },
        ]}
      />
    );
  };
  it('Save custom snapshot', async () => {
    const wrapperRender = renderer.create(<Demo />);
    expect(wrapperRender.toJSON()).toMatchSnapshot();
  });
  it('custom render props', async () => {
    const ChangeValue = 'btn-change-clik';

    const wrapper = mount(<Demo />);
    const changeBtn = wrapper.find('.change-btn').first();
    expect(changeBtn.getDOMNode()).toHaveProperty('disabled');
    // 取消禁用
    wrapper.find('.cancel-btn').first().simulate('click');
    await waitForComponentToPaint(wrapper);
    expect(changeBtn.getDOMNode()).toHaveProperty('disabled', false);
    // 修改数据
    changeBtn.simulate('click', { target: { value: ChangeValue } });
    await waitForComponentToPaint(wrapper);
    expect(onValuesChange).toBeCalledTimes(1);
    expect(changeBtn.text()).toBe(ChangeValue);
  });
});
