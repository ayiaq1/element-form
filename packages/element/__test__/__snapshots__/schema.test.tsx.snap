// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaForm Custom Save custom snapshot 1`] = `
<form
  className="ant-form ant-form-horizontal"
  disabled={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="diy"
        title="自定义组件"
      >
        自定义组件
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-horizontal ant-space-align-center"
            style={Object {}}
          >
            <div
              className="ant-space-item"
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <button
                className="ant-btn ant-btn-default change-btn"
                disabled={true}
                onClick={[Function]}
                type="button"
              />
            </div>
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn ant-btn-default cancel-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  取消禁用
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`SchemaForm Element Save type is custom snapshot 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="text"
        title="text"
      >
        text
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <span
            className="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line"
            onClick={null}
            style={
              Object {
                "WebkitLineClamp": undefined,
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="input"
        title="input"
      >
        input
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <span
            className="ant-input-affix-wrapper"
            onMouseUp={[Function]}
          >
            <input
              className="ant-input"
              hidden={null}
              maxLength={200}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              style={null}
              type="text"
              value=""
            />
            <span
              className="ant-input-suffix"
            >
              <span
                className="ant-input-clear-icon ant-input-clear-icon-hidden"
                onClick={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <span
                  aria-label="close-circle"
                  className="anticon anticon-close-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`SchemaForm Element Save type is row snapshot 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -8,
        "marginRight": -8,
      }
    }
  >
    <div
      className="ant-col ant-col-6"
      label="text"
      name="text"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
      type="text"
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="text"
            title="text"
          >
            text
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <span
                className="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line"
                onClick={null}
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-6"
      label="input"
      name="input"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
      type="input"
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input"
            title="input"
          >
            input
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <span
                className="ant-input-affix-wrapper"
                onMouseUp={[Function]}
              >
                <input
                  className="ant-input"
                  hidden={null}
                  maxLength={200}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  style={null}
                  type="text"
                  value=""
                />
                <span
                  className="ant-input-suffix"
                >
                  <span
                    className="ant-input-clear-icon ant-input-clear-icon-hidden"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    role="button"
                    tabIndex={-1}
                  >
                    <span
                      aria-label="close-circle"
                      className="anticon anticon-close-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="close-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
